### ISSUE TITLE
CollectionView adds only works when page is in focus on iOS

### ISSUE THREAD

**Issue by 1iveowl** [xamarin/Xamarin.Forms#8948]
### Description
Adding an item to the observable collection while the page with the CollectionView is in focus, seems to work just fine.

However, trying to add an item to the observable collection, while the page holding the CollectionView is **not i focus**, breaks the CollectionView on iOS, but not on Android. 

Specifically:
-  The CollectionView is not updated with the new item on iOS, when the page holding the CollectionView control returns into focus. 
- Also, any additional attempts to add an item to the observable collection, will throw this exception:

> Foundation.MonoTouchException  Message=Objective-C exception thrown.  Name: NSInternalInconsistencyException Reason: attempt to insert item 4 into section 0, but there are only 4 items in section 0 after the update...

### Steps to Reproduce
Please see the two videos of running the reproducing code on iOS and Android. Also see the link to the simple reproduce project, provided below.

### Expected Behavior
On Android it works as I would expect it to:
![Android - short](https://user-images.githubusercontent.com/12871120/71081298-361afe00-218f-11ea-91ef-82d88daed53b.gif)

### Actual Behavior
On iOS it does not work as I would expect it to work:
![iOS - Short](https://user-images.githubusercontent.com/12871120/71081319-429f5680-218f-11ea-8823-9b63a1938dbc.gif)

### Basic Information

- Version with issue: Xamarin.Forms 4.4.991265. I've tried with Xamarin Forms 4.3, but this seems to have the same issue.
- Last known good version: None that I know of.
- IDE: Visual Studio 16.4.2, .NET Core runtime 3.1.0
- Platform Target Frameworks: iOS, and Android.
  - iOS:  MDK 6.6.0.155, Xamarin.iOS 13.8.3.0, Xcode 11.3(15712),  
  - Android: 9.0, Xamarin Android 10.0.6.2
- Nuget Packages: Xamarin.Forms 4.4.0.991265, Xamarin.Essentials 1.3.1
- Affected Devices: iOS devices. I've tested both in the simulator and on physical iOS devices. It fails on both.

### Reproduction Link
https://github.com/1iveowl/XF.CollectionView.repro1

**Comment by rmarinho** [comment]
Works fine on Android, tried with 4.5.0.74-nighly still see the issue.

**Comment by Jared-Aus** [comment]
I can confirm that this is still an issue with 4.4.0 sr2, and for me adds to a long list of issues with CollectionView on iOS. I know my ItemsSource is okay since a test StackLayout using a BindableLayout on the same page updates perfectly.
Further to 1iveowl's detailed report, I can see that off-screen CollectionViews are somewhat handling the change notifications:

- Adding an item to a sorted collection where it wouldn't be last in the list seems to work, except the last item of the CollectionView then falls off.
- Deleting an item can appear to work flawlessly, unless the item template contains some static content and a noticeable background, in which case you can see a blank item at the end.
 
In summary it looks like the control is trying it's best to update the list but without the ability to change the size of the list.
At this point I'm not sure how to work around the issue short of re-architecting my app or replacing CollectionView with something that fully supports ObservableCollections.

**Comment by Jared-Aus** [comment]
Could I please request some urgency on this one? I've verified that this was working in 4.3.0.991221 and has been broken since 4.3.0.991250. This is a real blocker as I need some 4.4 features and recreating similar visuals with a ListView is not practical.
It looks like the changes to Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs for #7678 is to blame.
I haven't looked at 1iveowl's repo, but the easiest way for me to reproduce the issue is to use the Visual Studio project template for a XF Shell app and drop in a CollectionView next to the ListView that's bound to the same list. Adding an item updates the ListView but not the CollectionView. I don't think it's a good look when even a VS project template doesn't work with a CollectionView.

**Comment by anpin** [comment]
Does anyone found a workaround on this one?

**Comment by Jared-Aus** [comment]
> Does anyone found a workaround on this one?

I lost plenty of time trying to find one but without success. Possible options include rolling back to XF 4.3.0.991221, or switching to a ListView or BindingLayout. I went with the latter and bound my ObservableCollection to a StackLayout as I didn't need grouping, grid layouts or more than 20-30 items in the list.

**Comment by andy-vogan** [comment]
Unfortunately, I am currently seeing similar behavior with ListView when ItemsSource is set to an ObservableCollection and I modify that collection's items. I only touch the collection on the UI thread, however it still crashes in ListViewRenderer.cs when I modify the list. I will try rolling back the Xamarin.Forms version. After that I'm not sure what to try.

Either the person maintaining these renderer classes is breaking things really badly, or else perhaps iOS is changing the underlying behavior to make it more strict, resulting in crashing. Either way, my app is badly broken.

**Comment by hartez** [comment]
This is fixed in the latest stable version (5.0.0.1931).

**Comment by Eizzzi** [comment]
It was resolved in version 5.0.0.1931.
But I think the same issue is happening again in version 5.0.0.2012.
